security/pkg/k8s/chiron/utils.go

```diff
@@ -70,13 +76,30 @@ func getRandomCsrName(secretName, namespace string) string {
 	return name
 }
 
+func GetUniqueCsrName(csrCb CsrNameGenerator, client clientset.Interface, secretName, secretNamespace string) (string, error) {
```

Comment:
`GetUniqueCsrName` `GenCsrName` `getRandomCsrName` seems very confusing, can you unify them? UUID seems good to largely reduce the conflict. 

Retrying here to detect an existing name is useless, the period between this and when we create a csr, there is a chance a csr with the same name created.

Would recommend corporating generating the csr name with creating csr.

Reply:
Fixed this